### PR TITLE
[Keyboard] HS60 Refactor (Keymap restructure to be more standardized)

### PR DESCRIPTION
**Pull request by yiancar**
Hello all,
I was having a look at HS60 v2 in light of v3 being released and I realized that the keymaps were kind of not standard.

I think this way makes it a lot more coherent.

I have also split the config json files as a few people were getting confused, also hhkb was wrong.

Please let me know of further criticism/ideas. My motivation was keeping the default keymap being default no matter the layout (hhkb iso ansi)

Cheers!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Wilba6582** [comment]
Added bonus, the VIA enabled keymap is always called `via` so HS60 will be included in `make all:via` 

👍

**Comment by yiancar** [comment]
bump:)

**Comment by drashna** [comment]
To be blunt, I'm really not a fan of this change.  It's separating out the layout macros into individual keyboards (revisions/subfolders/etc), which is opposite of the direction that we have been moving.  

It makes the folder structure for the keyboards a lot more complicated to navigate, and harder to find the correct keyboard.  Which I feel is the wrong direction, too. 

The code changes themselves look good, but based on my listed concerns, I will not be approving the PR.  
I do not speak for QMK as a whole. This is just ***my*** opinion.  And if others approve it and want to merge it, I won't oppose it.

**Comment by yiancar** [comment]
My thinking is more in the line of.
You bought an ansi or iso or hhkb pcb. They are different physical products so for me I think having them in a subfolder makes more scene, as they are not the same pcb with multiple layouts.

**Comment by Wilba6582** [comment]
This structure makes more sense than before. @yiancar is dealing with three PCBs that have some commonality but each has a different and fixed layout. You can't use an ANSI layout macro on the ISO PCB, etc. They also need different vendor/product IDs so they can be distinguished by VIA.

The alternative (what was before) was loading too much overriding behaviour in the keymap so the PCBs could be distinguished.

While doing something like  `keyboards/hs60_v2_ansi`, `keyboards/hs60_v2_hhkb`. `keyboards/hs60_v2_iso` might be less folders, then the common code between these PCBs would need to go somewhere common, or be duplicated.

**Comment by yiancar** [comment]
Weekly Bump:) please give me some sort of direction on how to separate the physical pcbs!

**Comment by Wilba6582** [comment]
@yiancar the way you already did it

**Comment by noroadsleft** [comment]
I asked Skully about this a couple days ago, because I was of two minds about this change. Skully was of the opinion that the codebase *should* actually be separated out as this is, because it's three different PCBs, rather than one PCB that supports all three layouts.

IMO the bigger problem was this should have been originally submitted as three different folders, which I see as a failure on both @yiancar's end (bad submission) and ours (not getting caught and flagged on the original PR, and perhaps lack of visible documentation on this issue), and separating them out this way is a breaking change for the users, because this moves the keymap folders (the original `make` command won't work anymore).

In short, I approve of this PR in concept, but I'm not ready to merge it yet.

Tagging @stanrc85, @andys8 and @mcmadhatter so they're all aware of these changes.

**Comment by yiancar** [comment]
Yep I agree that it was a bad submision in the first place. I look at this PR and thinking that we as a community should be aware a bit more of individual physical pcbs and how we folder them. 

@noroadsleft Agreed about the macros, will be done asap!

**Comment by noroadsleft** [comment]
@yiancar I'm good with this PR as is; was just hoping to hear from stanrc85 and mcmadhatter.

**Comment by mcmadhatter** [comment]
Fine by me

**Comment by stanrc85** [comment]
I'm good with it

**Comment by noroadsleft** [comment]
Thanks guys.

@yiancar You've got a merge conflict now because #7911 was merged. Need a fix there.

**Comment by yiancar** [comment]
shall be f i x e d tonight:)

**Comment by yiancar** [comment]
@noroadsleft seems that everyone is happy now!

File: keyboards/hs60/v2/v2.c [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+
 #ifndef RGB_BACKLIGHT_HS60
 #error RGB_BACKLIGHT_M60_A not defined, recheck config.h
```
Comment (fauxpark) [inline keyboards/hs60/v2/v2.c]:
```suggestion
#error RGB_BACKLIGHT_HS60_A not defined, recheck config.h
```

Comment (yiancar) [reply keyboards/hs60/v2/v2.c]: wow! that was there a long time

**Comment by noroadsleft** [comment]
Thanks! :tada:
